pkg/clusteragent/admission/mutate/autoinstrumentation/target_mutator.go

```diff
@@ -170,6 +173,17 @@ func (m *TargetMutator) MutatePod(pod *corev1.Pod, ns string, _ dynamic.Interfac
 	}
 	extracted := m.core.initExtractedLibInfo(pod).withLibs(target.libVersions)
 
+	// If the user doesn't set ddTraceVersions, we should use language detection.
+	var usedLanguageDetection bool
+	if target.languageDetectionEligible {
```

Comment:
I think this would override pinned versions, we should consider splitting into two PRs so we can get the service name fix out.

Reply:
Or do we not use `instrumentation.libVersions` when targets are set? I don't remember

Reply:
Checked, we don't do `libVersions`